hbase-client/src/main/java/org/apache/hadoop/hbase/filter/PrefixFilter.java

```diff
@@ -142,6 +155,31 @@ boolean areSerializedFieldsEqual(Filter o) {
     return Bytes.equals(this.getPrefix(), other.getPrefix());
   }
 
+  @Override
+  public Cell getNextCellHint(Cell cell) {
+    byte[] hintBytes;
+    if (reversed) {
+      // On reversed scan hint should be the prefix with last byte incremented
+      hintBytes = increaseLastNonMaxByte(this.prefix);
```

Comment:
Thanks. To be completely honest I am wondering on this a bit. :thinking: 

If I understand it correctly, when the filter returns the hint, then we will be "navigated" to a row which is just before the first real match. Then, on advancing to the next row, the first matching row will be found.

So as I understand, there will be no cells between the hint and the first real match. If this is true, does it make sense to pre-compute this hint?

Reply:
You have a point.
However, the API does not explicitly state that it will honor the hint, so I think it's better to pre-compute it, in case we have to return the hint several times.

We lose a few cycles if we never need to return a hint, but we win a lot of cycles if we have to return it a lot of times.

Reply:
I thought about this some more:

At least one case when the HINT is not honored is FilterListWithOR , which will use the closest HINT provided by any of the filters, so if another filter returns a hint that is closer than the one PrefixFilter provides, it is going to be evaluated, possibly a lot of times.
Pre-computing the hint is definitely the most optimal solution.

Reply:
Thanks for the added explanation. Makes perfect sense. I changed the reverse hint to be pre-computed.